File: lib/features/chat/models/chat_message.dart
enum SenderType { shopper, customer }

enum MessageType { text, image, priceCard }

class ChatMessage {
  const ChatMessage({
    required this.id,
    required this.sender,
    required this.type,
    required this.time,
    this.text,
    this.imageUrl,
    this.priceCardData,
    this.replyToText,
  });

  final String id;
  final SenderType sender;
  final MessageType type;
  final DateTime time;
  final String? text;
  final String? imageUrl;
  final PriceCardData? priceCardData;
  /// Snippet of the message being replied to (local only, not persisted).
  final String? replyToText;
}

class PriceCardData {
  const PriceCardData({
    required this.itemName,
    required this.imageUrl,
    required this.price,
  });

  final String itemName;
  final String imageUrl;
  final double price;
}

File: backend/src/SwiftShopper.Domain/Entities/ChatMessage.cs
namespace SwiftShopper.Domain.Entities;

public class ChatMessage
{
    public required string Id { get; init; }

    public required string OrderId { get; init; }

    public required string Sender { get; init; }

    public required string Type { get; init; }

    public string? Text { get; init; }

    public string? ImageUrl { get; init; }

    /// <summary>Populated when Type == "price-card".</summary>
    public PriceCardData? PriceCard { get; init; }

    public DateTimeOffset SentAt { get; init; }
}

/// <summary>
/// Structured payload attached to price-card chat messages.
/// Allows the customer to see exactly what the shopper is proposing.
/// </summary>
public class PriceCardData
{
    public required string ItemName { get; init; }

    public string Quantity { get; init; } = string.Empty;

    public decimal OldPrice { get; init; }

    public decimal NewPrice { get; init; }
}
